query/src/test/java/io/keen/client/java/PersistentAnalysisTest.java

```diff
@@ -0,0 +1,554 @@
+
+import io.keen.client.java.http.Response;
+import io.keen.client.java.result.FunnelResult;
+import io.keen.client.java.result.MultiAnalysisResult;
+import io.keen.client.java.result.QueryResult;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test the Persistent Analysis query functionality, which should include Saved/Cached Queries and
+ * Cached Datasets.
+ *
+ * @author masojus
+ */
+public class PersistentAnalysisTest extends KeenQueryTestBase {
+    private static final String TEST_RESOURCE_NAME = "some_name";
+
+    private SavedQueries savedQueryApi = null;
+
+    @Before
+    public void persistentAnalysisTestSetup() {
+        savedQueryApi = queryClient.getSavedQueriesInterface();
+    }
+
+    @After
+    public void persistentAnalysisTestTeardown() {
+        savedQueryApi = null;
+    }
+
+    // Make sure result, if provided, is wrapped in root object/array brackets, if not scalar.
+    private String getFakeSavedQueryDefinition(String result, boolean isGroupBy, boolean isInterval, boolean isMultiAnalysis, boolean isFunnel) {
+        // A few of the keys one would get back. We don't really check the return values in most of
+        // these tests because we aren't parsing them for the most part, except the query result.
+        String analysisType;
+
+        if (isMultiAnalysis) {
+            analysisType = "multi_analysis";
+        } else if (isFunnel) {
+            analysisType = "funnel";
+        } else {
+            analysisType = "sum";
+        }
+
+        String definition = "{" +
+                    "\"query_name\": \"" + TEST_RESOURCE_NAME + "\"," +
+                    "\"refresh_rate\": 0," +
+                    "\"query\": {" +
+                        "\"analysis_type\": \"" + analysisType + "\"," +
+                        "\"target_property\": \"someProp\"," +
+                        "\"timeframe\": \"this_month\"" + (!isGroupBy ? "" : ", " +
+                        "\"group_by\": [\"category\"]") + (!isInterval ? "" : ", " +
+                        "\"interval\": \"daily\"") +
+                    "}" + (null == result ? "" : ", " +
+                    "\"result\": " + result) +
+                "}";
+
+        return definition;
+    }
+
+    private String getFakeSavedQueryDefinition(String result) {
+        return getFakeSavedQueryDefinition(result, false, false, false, false);
+    }
+
+    private static void validatePersistentAnalysisRequiredFields(ObjectNode requestNode) {
+        // Should have 'refresh_rate' and 'query' top-level keys, at least.
+        assertTrue("Missing required top-level fields.", 2 <= requestNode.size());
+
+        assertTrue(requestNode.hasNonNull(KeenQueryConstants.REFRESH_RATE));
```

Comment:
Yes, for now. It seems like a bug, but I'm not sure when they'll get to fixing it, and it's just as easy for us to pass it along.